indykite/resource_consent_test.go

```diff
@@ -0,0 +1,401 @@
+// limitations under the License.
+
+package indykite_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"sync"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/indykite/indykite-sdk-go/config"
+	configpb "github.com/indykite/indykite-sdk-go/gen/indykite/config/v1beta1"
+	configm "github.com/indykite/indykite-sdk-go/test/config/v1beta1"
+	"github.com/pborman/uuid"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/indykite/terraform-provider-indykite/indykite"
+	"github.com/indykite/terraform-provider-indykite/indykite/test"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+)
+
+var _ = Describe("Resource Consent config", func() {
+	const resourceName = "indykite_consent.wonka"
+	var (
+		mockCtrl         *gomock.Controller
+		mockConfigClient *configm.MockConfigManagementAPIClient
+		provider         *schema.Provider
+		mockedBookmark   string
+	)
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(TerraformGomockT(GinkgoT()))
+		mockConfigClient = configm.NewMockConfigManagementAPIClient(mockCtrl)
+
+		// Bookmark must be longer than 40 chars - have just 1 added before the first write to test all cases
+		mockedBookmark = "for-consent" + uuid.NewRandom().String()
+		bmOnce := &sync.Once{}
+		provider = indykite.Provider()
+		cfgFunc := provider.ConfigureContextFunc
+		provider.ConfigureContextFunc =
+			func(ctx context.Context, data *schema.ResourceData) (any, diag.Diagnostics) {
+				client, _ := config.NewTestClient(ctx, mockConfigClient)
+				ctx = indykite.WithClient(ctx, client)
+				i, d := cfgFunc(ctx, data)
+				// ConfigureContextFunc is called repeatedly, add initial bookmark just once
+				bmOnce.Do(func() {
+					i.(*indykite.ClientContext).AddBookmarks(mockedBookmark)
+				})
+				return i, d
+			}
+	})
+
+	It("Test all CRUD", func() {
+		consentConfigResp := &configpb.ReadConfigNodeResponse{
+			ConfigNode: &configpb.ConfigNode{
+				CustomerId:  customerID,
+				AppSpaceId:  appSpaceID,
+				Id:          sampleID,
+				Name:        "wonka-consent-config",
+				DisplayName: "Wonka Consent for chocolate receipts",
+				CreateTime:  timestamppb.Now(),
+				UpdateTime:  timestamppb.Now(),
+				Config: &configpb.ConfigNode_ConsentConfig{
+					ConsentConfig: &configpb.ConsentConfiguration{
+						Purpose:        "To allow Wonka to store and process chocolate receipts",
+						ApplicationId:  applicationID,
+						RevokeAfterUse: true,
+						ValidityPeriod: 86400,
+						DataPoints:     []string{"{\"returns\": [{\"properties\": [\"name\", \"location\"]}]}"},
```

Comment:
Suggestion: If you use ``` ` ``` you don't have to escape every `"`"

Reply:
The use of single quotes is not allowed in terraform files

Reply:
1. This is Go code, not Terraform code. You can use ``` ` ``` here
2. In Terraform code you can use `<<-EOT` or 

```
jsonencode({
	"kid": "abc",
	"use": "sig",
	"alg": "RS256",
	"n": "--nothing-real-just-random-xyqwerasf--",
	"kty": "RSA"
}),
```

Reply:
Ah yes sorry I  miss red the file, I  thought it was the example file, I will also fix the example file with jsonencode just to be readable